internal/service/controlplaneipaddresses/model_test.go

```diff
@@ -0,0 +1,112 @@
+package controlplaneipaddresses_test
+
+import (
+	"context"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-framework/types/basetypes"
+	"github.com/mongodb/terraform-provider-mongodbatlas/internal/common/conversion"
+	"github.com/mongodb/terraform-provider-mongodbatlas/internal/service/controlplaneipaddresses"
+	"github.com/stretchr/testify/assert"
+	"go.mongodb.org/atlas-sdk/v20231115014/admin"
+)
+
+type sdkToTFModelTestCase struct {
+	SDKResp         *admin.ControlPlaneIPAddresses
+	expectedTFModel *controlplaneipaddresses.TFControlPlaneIpAddressesModel
+	name            string
+}
+
+func TestControlPlaneIpAddressesSDKToTFModel(t *testing.T) {
+	testCases := []sdkToTFModelTestCase{
```

Comment:
knit: it's a bit better to have a map string to the result instead of having name inside the struct

Reply:
Not a strong opinion but I do find leaving the `name` property to be more clear than using the key of the map, but happy to adjust so we remain consistent in our codebase.

Reply:
synced offline, adjusted fixture for these tests and our scaffolding template.